.tito/lib/git_annex_spec_builder.py

```diff
@@ -0,0 +1,168 @@
+        self.tgz_dir = tgz_base
+        self.artifacts = []
+
+        self.rpmbuild_gitcopy = os.path.join(self.rpmbuild_sourcedir,
+                self.tgz_dir)
+
+        # Used to make sure we only modify the spec file for a test build
+        # once. The srpm method may be called multiple times during koji
+        # releases to create the proper disttags, but we only want to modify
+        # the spec file once.
+        self.ran_setup_test_specfile = False
+
+        # NOTE: These are defined later when/if we actually dump a copy of the
+        # project source at the tag we're building. Only then can we search for
+        # a spec file.
+        self.spec_file_name = None
+        self.spec_file = None
+
+        # Set to path to srpm once we build one.
+        self.srpm_location = None
+
+    def _get_build_version(self):
+        self.spec_version = ''
+        self.spec_release = ''
+        build_version = get_spec_version_and_release(
+            self.start_dir,
+            find_spec_like_file(self.start_dir)
+        )
+        self.build_tag = build_version
+        return build_version
+
+    def _setup_sources(self):
+        """
+        Create a copy of the git source for the project at the point in time
+        our build tag was created.
+
+        Created in the temporary rpmbuild SOURCES directory.
+        """
+        self._create_build_dirs()
+
+        debug('SETUP SOURCES')
+        working_path = os.getcwd() if self.relative_project_dir in os.path.join(os.getcwd(), '') else os.path.join(os.getcwd(), self.relative_project_dir)
+        debug("working_path: %s" % working_path)
+
+        for directory, unused, filenames in os.walk(working_path):
+            debug('WALK')
+            dir_artifacts_with_path = [os.path.join(directory, f) for f in filenames]
+
+            debug(dir_artifacts_with_path)
+            for artifact in dir_artifacts_with_path:
+                debug("  Copying source file %s" % artifact)
+                if os.path.isfile(artifact):
+                    shutil.copy(artifact, self.rpmbuild_gitcopy)
+                    shutil.copy(artifact, self.rpmbuild_sourcedir)
+
+        # NOTE: The spec file we actually use is the one exported by git
+        # archive into the temp build directory. This is done so we can
+        # modify the version/release on the fly when building test rpms
+        # that use a git SHA1 for their version.
+        self.spec_file_name = os.path.basename(find_spec_like_file(self.rpmbuild_sourcedir))
+        self.spec_file = os.path.join(
+            self.rpmbuild_sourcedir, self.spec_file_name)
+
+        self.old_cwd = os.getcwd()
+        if self.relative_project_dir not in os.path.join(os.getcwd(), ''):
+            os.chdir(os.path.join(self.old_cwd, self.relative_project_dir))
+
+        # NOTE: 'which' may not be installed... (docker containers)
+        (status, output) = getstatusoutput("which git-annex")
```

Comment:
Python 3.3+ has ```shutil.which```. Time to upgrade ;)

Reply:
I took this from some tito code. If we want to lock this to python3 we can.

Reply:
It was a bit sarcastic since EL7 doesn't include python3 which might be an issue.